elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+int lowRange = 0;
+int highRange = 15;
+Map<String, String> filtering = new HashMap<>();
+filtering.put("location", "/home/audit/logs");
+filtering.put("suffix", ".yyyy-MM-dd_HHmm");
+AuditEndpoint endpoint = PeriodicRotatingFileAuditEndpoint.builder()
+        .setTimeZone(UTC)
+        .setSuffix(".yyyy-MM-ww-dd-a-HH_mm")
+        .setLocation(logFile)
+        .setClock(clock)
+        .setFiltering(criteria, lowRange, highRange, filtering)
+        .build();
+----
+[source,xml]
+----
+... criteria="facility" values="0,15" filtering="location='/home/audit/logs',suffix='.yyyy-MM-dd_HHmm'" ...
+----
+
+* Logging should be done by using the proposal API as described in https://github.com/wildfly/wildfly-proposals/pull/181[WFCORE-4355]
+** A StdoutAuditEndpoint should be added to WildFly-Core that uses this API to log an event to Stdout
+** An AysncAuditEndpoint should be added to WildFly-Core that uses this API to log events to a queue before they're writing to Stdout
+** JsonSecurityEventFormatter should be moved to WildFly-Core and modified to use the JsonFormatter in this API
+** The current FileAuditEndpoints, FileAuditEndpoint, PeriodicRotatingFileAuditEndpoint, and SizeRotatingFileAuditEndpoint
+should be moved to WildFly-Core
+*** The API should be extended to provide the capability necessary for these Endpoints
+
+|===
+|Criteria |Possible Values |Method of Input
+
+| Type
+| Assertion, Audit Information (Syslog Event, Audit Log Enabled, etc), Authentication, Failure, Login, Permission, Successful, Unknown, Validation
+| Java String List for programmatically, comma separated list for markup
+
+| Facility
+| Values ranging from 0 to 23, as described in https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+| Two int values, first being the lower value and second being the higher value, for programmatically, lower value and higher value separated by a comma for markup
+
+| Severity
+| Values ranging from 0 to 7, as described in https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+| Two int values, first being the lower value and second being the higher value, for programmatically, lower value and higher value separated by a comma for markup
+
+| Priority
+| Values ranging from 0 to 191, calculated by multiplying the Facility value by 8 and adding the Severity value, as described in https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+| Two int values, first being the lower value and second being the higher value, for programmatically, lower value and higher value separated by a comma for markup
+
+| Level
+| FATAL, ERROR, WARN, INFO, DEBUG, TRACE
+| Java String List for programmatically, comma separated list for markup
+
+|===
+
+* Additional audit events should be added for feature parity with Legacy Security Audit Logging. The following tables makes a comparison of Legacy Security Audit Logging features included in, not included in, and if the feature should be added to Elytron Audit Logging:
```

Comment:
What do you mean by adding event? Where they will be added?

Reply:
By adding events I mean adding additional options for events that can be logged and they would coexist with the other event types in org.wildfly.security.auth.server.event and be added to the formatters

Reply:
I am sorry, but I still don't understand what does that mean from users point of view.

Reply:
For the user, it would just affect the types of events that get logged when they look at their log. It would also tie into the filtering system above in having more options for the types of events that the user wishes to log/not log